controllers/cards.js

```diff
@@ -1,47 +1,56 @@
 const Card = require('../models/card');
+const ErrorMessage = require('../assets/error-message');
 
-// возвращает все карточки
-const cardsGet = (req, res) => {
-    Card.find({})
-        .then(cards => res.send(cards))
-        .catch(err => res.status(500).send({ message: 'Произошла ошибка в выводе списка карточек', error: err }));
-};
+module.exports = {
+    // возвращает все карточки
+    cardsGet: (req, res) => {
+        Card.find({})
+            .then(cards => res.send(cards))
+            .catch(err => new ErrorMessage(err, 'cardsGet', res));
+    },
 
-// создаёт карточку
-const cardPost = (req, res) => {
-    const {name, link} = req.body;
-    Card.create({name, link, owner: req.user._id})
-        .then(card => res.send(card))
-        .catch(err => res.status(500).send({ message: 'Произошла ошибка в создании новой карточки', error: err }));
-};
+    // создаёт карточку
+    cardPost: (req, res) => {
+        const {name, link} = req.body;
+        Card.create({name, link, owner: req.user._id})
+            .then(card => res.send(card))
+            .catch(err => new ErrorMessage(err, 'cardPost', res));
+    },
 
-// удаляет карточку по идентификатору
-const cardDelete = (req, res) => {
-    Card.findByIdAndRemove(req.params.cardId)
-        .then(card => res.send(card))
-        .catch(err => res.status(500).send({ message: 'Произошла ошибка в удалении карточки', error: err }));
-};
+    // удаляет карточку по идентификатору
+    cardDelete: (req, res) => {
+        Card.findById(req.params.id)
+            .orFail()
+            .then(card => {
+                if (!card) return Promise.reject({name: 'DocumentNotFoundError'});
+                if (!card.owner.equals(req.user._id)) return Promise.reject({name: 'ForbiddenError'});
 
-// поставить лайк карточке
-const cardLikePut = (req, res) => {
-    Card.findByIdAndUpdate(
-        req.params.cardId,
-        { $addToSet: {likes: req.user._id} }, // добавить _id в массив, если его там нет
-        { new: true },
-    )
-    .then(card => res.send(card))
-    .catch(err => res.status(500).send({ message: 'Произошла ошибка в добавлении лайка карточки', error: err }));
-};
+                return Card.deleteOne(card).then(card => res.send(card));
```

Comment:
Можно лучше: возвращать удаленную карточку при выполнении удаления.
Сейчас вместо данных о карточке возвращается информация какое количество записей было удалено:
```json
{
    "n": 1,
    "ok": 1,
    "deletedCount": 1
}
```
Для этого можно вернуть `card`, который был результатом выполнения `.findById()`. Те:
```suggestion
                return Card.deleteOne(card).then(() => res.send(card));
```